src/main/scala/tests/visibility.scala

```diff
@@ -0,0 +1,57 @@
+package tests
+package visibility
+
+private object PrivateTopLevelObject //unexpected
+
+private[tests] object PrivateInOuterPackageTopLevelObject //unexpected
+
+private[visibility] object PrivateInInnerPackageTopLevelObject //unexpected
+
+private[this] object LocallyPrivateTopLevelObject //unexpected
+
+protected object ProtectedTopLevelObject
+
+protected[tests] object ProtectedInOuterPackageTopLevelObject //unexpected
+
+protected[visibility] object ProtectedInInnerPackageTopLevelObject //unexpected
+
+protected[this] object LocallyProtectedTopLevelObject //unexpected
+
+private def PrivateTopLevelMethod: Int //unexpected
+    = 1
+
+protected def ProtectedTopLevelMethod: Int
+    = 1
+
+class InClassVisibility()
+{
+    private def privateMethod: Int //unexpected
+      = ???
+
+    private[tests] def privateInOuterPackageMethod: Int //unexpected
+      = ???
+
+    private[visibility] def privateInInnerPackageMethod: Int //unexpected
+      = ???
+
+    private[InClassVisibility] def privateInClassMethod: Int //unexpected
+      = ???
+
+    private[this] def locallyPrivateMethod: Int //unexpected
+      = ???
+
+    protected def protectedMethod: Int
+      = ???
+
+    protected[tests] def protectedInOuterPackageMethod: Int //unexpected
+      = ???
+
+    protected[visibility] def protectedInInnerPackageMethod: Int //unexpected
+      = ???
+
+    protected[InClassVisibility] def protectedInClassClassMethod: Int //unexpected
```

Comment:
We actually want to document at least some class-local definitions; in particular, this one. `protected[InClassVisibility]` inside `InClassVisibility` is the same as `protected`, and we show those. In principle, I think we want to document class-local definitions if the class they're local in is itself documented, but if that's too difficult to implement, we can just always document them.

Reply:
And should they be displayed as `protected` or `protected[EnclosingClass]` inside the documentation?

Reply:
It's fine to display them as `protected[EnclosingClass]` - this is what the user entered and trying to special-case how it is displayed isn't especially important.